extendr-api/src/thread_safety.rs

```diff
@@ -77,23 +77,17 @@ where
 {
     match std::panic::catch_unwind(f) {
         Ok(res) => res,
-        Err(_) => {
-            unsafe {
-                libR_sys::Rf_error(err_str.as_ptr() as *const std::os::raw::c_char);
-            }
-            unreachable!("handle_panic unreachable")
-        }
+        Err(_) => unsafe { libR_sys::Rf_error(err_str.as_ptr() as *const std::os::raw::c_char) },
     }
 }
 
 static mut R_ERROR_BUF: Option<std::ffi::CString> = None;
 
-pub fn throw_r_error<S: AsRef<str>>(s: S) {
```

Comment:
Isn't `never` unstable? https://doc.rust-lang.org/std/primitive.never.html

Reply:
Yes, but this is allowed, see this https://doc.rust-lang.org/reference/types/never.html

Reply:
Okay, so it's considered a "diverging function" which is what we called these functions before `never` was introduced and subsequently un-introduced: https://doc.rust-lang.org/rust-by-example/fn/diverging.html.